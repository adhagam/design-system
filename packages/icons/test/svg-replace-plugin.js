import { transform } from '@babel/core'
import replaceSvgPlugin from '../plugins/svg-dynamic-title-plugin'

const testPlugin = (code, options) => {
  const result = transform(code, {
    plugins: ['@babel/plugin-syntax-jsx', [replaceSvgPlugin, options]],
    configFile: false
  })

  return result.code
}

describe('replace svg plugin', () => {
  it('replace svg with Svg', () => {
    expect(testPlugin('<svg></svg>')).toMatchSnapshot(`'<Svg></Svg>'`)
  })

  it('replace svg with Svg when no closing element is provided', () => {
    expect(testPlugin('<svg />')).toMatchSnapshot(`'<Svg />'`)
  })
})
