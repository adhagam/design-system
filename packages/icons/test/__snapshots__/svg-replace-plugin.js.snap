// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replace svg plugin replace svg with Svg when no closing element is provided: '<Svg />' 1`] = `"<svg />;"`;

exports[`replace svg plugin replace svg with Svg: '<Svg></Svg>' 1`] = `"<svg>{!!title && (<title id={titleId}>{title}</title>)}{!!desc && (<desc id={descId}>{desc}</desc>)}</svg>;"`;
